mobile/src/actions/index.js

```diff
@@ -1 +1,44 @@
+import { Cozy, MemoryStorage } from 'cozy-client-js'
+
 export const SETUP = 'SETUP'
+export const SET_URL = 'SET_URL'
+
+export const setupServerUrl = (router, location) => {
+  return async (dispatch, getState) => {
+    let cozy = new Cozy({
+      cozyURL: `http://${getState().mobile.serverUrl}`,
+      oauth: {
+        storage: new MemoryStorage(),
+        clientParams: {
+          redirectURI: 'http://localhost',
+          softwareID: 'mysoftware',
+          clientName: 'Great Mobile App',
+          scopes: ['files/images:read']
+        },
+        onRegistered: (client, url) => {
+          if (window.cordova && window.cordova.InAppBrowser) {
+            let inAppBrowser = window.cordova.InAppBrowser.open(url, '_blank', 'location=yes,hidden=yes')
+            inAppBrowser.addEventListener('loadstop', (e) => {
+              inAppBrowser.show()
+
+              var url = e.url
+              var code = /\?access_code=(.+)$/.exec(url)
+              var error = /\?state=(.+)$/.exec(url)
+
+              if (code || error) {
+                inAppBrowser.close()
+                dispatch({ type: SETUP })
+                if (location.state && location.state.nextPathname) {
+                  router.replace(location.state.nextPathname)
+                } else {
+                  router.replace('/')
+                }
+              }
+            })
+          }
+        }
+      }
+    })
+    cozy.authorize()
```

Comment:
In that case, how could I know if the client is authorized or not? In order to manage the in-app browser opening the authorization page.